src/coreclr/jit/emitxarch.cpp

```diff
@@ -9769,38 +9773,63 @@ const char* emitter::emitRegName(regNumber reg, emitAttr attr, bool varName)
 
     const char* rn = emitComp->compRegVarName(reg, varName);
 
-#ifdef TARGET_AMD64
     char suffix = '\0';
 
     if (isMaskReg(reg))
     {
         return rn;
     }
 
+#ifdef TARGET_X86
+    assert(strlen(rn) >= 3);
+#endif // TARGET_X86
+
     switch (EA_SIZE(attr))
     {
         case EA_64BYTE:
-            return emitZMMregName(reg);
+        {
+            if (IsXMMReg(reg))
+            {
+                return emitZMMregName(reg);
+            }
+            break;
```

Comment:
While we would ideally pass in the "correct" attribute all the time, we have a number of instructions where that doesn't happen and where we end up with things like `xrdx` or `yrdx`.

Doing the `IsXMMReg(reg)` check at least ensures we don't do that and simply get `edx`/`rdx` instead for general purpose registers. This helps fix a number of diffs that were printing slightly incorrectly and gives us an easier way to identify operands with the "wrong" attribute size being passed in.